frontend/javascripts/oxalis/view/left-border-tabs/histogram_view.js

```diff
@@ -185,6 +193,80 @@ class Histogram extends React.PureComponent<HistogramProps, HistogramState> {
     500,
   );
 
+  getCuboidForViewport = async (viewport: OrthoView, layerName: string) => {
+    const state = Store.getState();
+
+    const [curX, curY, curZ] = Dimensions.transDim(
+      Dimensions.roundCoordinate(getPosition(state.flycam)),
+      viewport,
+    );
+    const [halfViewportExtentX, halfViewportExtentY] = getHalfViewportExtentsFromState(
+      state,
+      viewport,
+    );
+
+    const min = Dimensions.transDim(
+      V3.sub([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 0]),
+      viewport,
+    );
+    const max = Dimensions.transDim(
+      V3.add([curX, curY, curZ], [halfViewportExtentX, halfViewportExtentY, 1]),
+      viewport,
+    );
+
+    const resolutionIndex = getRequestLogZoomStep(state);
+    const cuboid = await api.data.getDataFor2DBoundingBox(layerName, { min, max }, resolutionIndex);
+    return cuboid;
+  };
+
+  getClippingValues = async (layerName: string, threshold: number = 0.05) => {
+    const cuboidXY = await this.getCuboidForViewport(OrthoViews.PLANE_XY, layerName);
+    const cuboidXZ = await this.getCuboidForViewport(OrthoViews.PLANE_XZ, layerName);
+    const cuboidYZ = await this.getCuboidForViewport(OrthoViews.PLANE_YZ, layerName);
+
+    const { elementClass } = getLayerByName(Store.getState().dataset, layerName);
+    const [TypedArrayClass] = getConstructorForElementClass(elementClass);
+    const cuboidData = new TypedArrayClass(cuboidXY.length + cuboidXZ.length + cuboidYZ.length);
+
+    cuboidData.set(cuboidXY);
+    cuboidData.set(cuboidXZ, cuboidXY.length);
+    cuboidData.set(cuboidYZ, cuboidXY.length + cuboidXZ.length);
+
+    const valueCounts = {};
+    let maxCount = 0;
+    for (let i = 0; i < cuboidData.length; i++) {
+      if (cuboidData[i] !== 0 && cuboidData[i] in valueCounts) {
+        valueCounts[cuboidData[i]] += 1;
+        if (maxCount < valueCounts[cuboidData[i]]) maxCount = valueCounts[cuboidData[i]];
+      } else valueCounts[cuboidData[i]] = 1;
+    }
+
+    let lowClip = 255;
+    let highClip = 1;
+    const threshValue = (threshold * maxCount).toFixed();
+    for (let i = 0; i < cuboidData.length; i++) {
+      if (valueCounts[cuboidData[i]] > threshValue) {
+        if (cuboidData[i] !== 0 && cuboidData[i] < lowClip) lowClip = cuboidData[i];
+        if (cuboidData[i] > highClip) highClip = cuboidData[i];
+      }
+    }
+
+    if (lowClip > highClip) return [highClip, lowClip];
+    return [lowClip, highClip];
+  };
+
+  clipHistogram = async (isInEditMode: boolean, layerName: string) => {
+    const [lowClip, highClip] = await this.getClippingValues(layerName);
+    if (!isInEditMode) {
+      this.onThresholdChange([lowClip, highClip]);
+    } else {
+      this.onThresholdChange([lowClip, highClip]);
+      this.setState({ currentMin: lowClip, currentMax: highClip });
+      this.updateMinimumDebounced(lowClip, layerName);
```

Comment:
The debounced version is meant to collapse multiple function calls if they are shortly fired after each other (useful for mouse events which can be triggered frequently in a short time). Since it always introduces an artificial delay (in this case 500 ms), I'd use `this.props.onChangeLayer(layerName, "min", value)` instead (analoguous for the maximum).